Apps/Patient/src/Services/PatientDataService.cs

```diff
@@ -15,63 +15,119 @@
+            Guid pid = await this.ResolvePidFromHdid(query.Hdid).ConfigureAwait(true);
+            PatientFile? file = (await this.patientDataRepository.Query(new PatientDataAccess.PatientFileQuery(pid, query.FileId), ct).ConfigureAwait(true)).Items.FirstOrDefault() as PatientFile;
 
             return file != null
                 ? new PatientFileResponse(file.Content, file.ContentType)
                 : null;
         }
 
-        private async Task<Guid> ResolvePidFromHdid(string patientHdid) =>
-            (await this.personalAccountsService.GetPatientAccountAsync(patientHdid).ConfigureAwait(true)).PatientIdentity.Pid;
+        private static IEnumerable<HealthServiceCategory> ExtractToHealthServiceCategoryArray(IEnumerable<PatientDataType> patientDataType)
+        {
+            foreach (PatientDataType pdt in patientDataType)
+            {
+                HealthServiceCategory? hsc = pdt switch
+                {
+                    PatientDataType.OrganDonorRegistrationStatus => HealthServiceCategory.OrganDonor,
+                    _ => null,
+                };
+                if (hsc != null)
+                {
+                    yield return hsc.Value;
+                }
+            }
+        }
 
-        private PatientData MapToPatientData(HealthData healthData) =>
-            healthData switch
+        private static IEnumerable<HealthDataCategory> ExtractToHealthDataCategoryArray(IEnumerable<PatientDataType> patientDataType)
+        {
+            foreach (PatientDataType pdt in patientDataType)
             {
-                OrganDonorRegistration hd => new OrganDonorRegistrationData(hd.Status, hd.StatusMessage, hd.RegistrationFileId),
-                _ => throw new NotImplementedException($"{healthData.GetType().Name} is not mapped to {nameof(PatientData)}"),
-            };
+                HealthDataCategory? hdc = pdt switch
+                {
+                    PatientDataType.DiagnosticImaging => HealthDataCategory.DiagnosticImaging,
+                    _ => null,
+                };
+                if (hdc != null)
+                {
+                    yield return hdc.Value;
+                }
+            }
+        }
 
-        private HealthServiceCategory MapToHealthServiceCategory(PatientDataType patientDataType) =>
-            patientDataType switch
+        private async Task<Guid> ResolvePidFromHdid(string patientHdid)
+        {
+            return (await this.personalAccountsService.GetPatientAccountAsync(patientHdid).ConfigureAwait(true)).PatientIdentity.Pid;
+        }
+
+        private async Task<PatientDataQueryResult> HandleServiceQuery(Guid pid, PatientDataQuery query, CancellationToken ct)
+        {
+            IEnumerable<HealthServiceCategory> categories = ExtractToHealthServiceCategoryArray(query.PatientDataTypes).ToArray();
```

Comment:
The ToArray() appears unnecessary.  The method name could probably also be updated to something like ExtractHealthServiceCategories.